tests/greedy-url/pass_ms.js

```diff
@@ -0,0 +1,14 @@
+const getURL = (data) => data.match(/https*:\/\/\S+/g)
+
+const greedyQuery = (data) => data.match(/https*:\/\/\S+\?(\S+=\S+\&*){3,}/g)
+
+const notSoGreedy = (data) => {
+  let res = data.match(/https*:\/\/\S+\?(\w+=[^&\r\n\t\f\v ]+&*){2,3}/g)
+  for (let i = 0; i < res.length; i++) {
+    if (res[i].charAt(res[i].length - 1) == '&') {
+      res.splice(i, 1)
```

Comment:
if you want to remove elements from an array: use `filter`, if you want to get a slice of it use `slice`
but avoid `splice` when ever you can, you always have a better way in JS (your solution is still completely valid, i'm just saying that for you)